src/optimize/base_optimizer.js

```diff
@@ -105,61 +116,106 @@ export default class BaseOptimizer {
-          { test: /[\/\\]src[\/\\](core_plugins|ui)[\/\\].+\.js$/, loader: loaderWithSourceMaps('rjs-repack-loader') },
+        rules: [
+          {
+            test: /\.less$/,
+            use: getStyleLoaders(
+              ['less-loader'],
+              [{
+                loader: 'cache-loader',
+                options: {
+                  cacheDirectory: resolve(cacheDirectory, 'less'),
+                }
+              }]
+            ),
+          },
+          {
+            test: /\.css$/,
+            use: getStyleLoaders(),
+          },
+          {
+            // TODO: this doesn't seem to be used, remove?
+            test: /\.jade$/,
+            loader: 'jade-loader'
+          },
+          {
+            test: /\.(html|tmpl)$/,
+            loader: 'raw-loader'
+          },
+          {
+            test: /\.png$/,
+            loader: 'url-loader'
+          },
+          {
+            test: /\.(woff|woff2|ttf|eot|svg|ico)(\?|$)/,
+            loader: 'file-loader'
+          },
           {
             test: /\.js$/,
-            exclude: babelExclude.concat(this.env.noParse),
-            loader: 'babel-loader',
-            query: {
-              presets: [
-                require.resolve('../babel-preset/webpack')
-              ]
-            }
+            exclude: BABEL_EXCLUDE_RE.concat(this.env.noParse),
+            use: [
+              {
+                loader: 'cache-loader',
+                options: {
+                  cacheDirectory: resolve(cacheDirectory, 'babel'),
+                }
+              },
+              {
+                loader: 'babel-loader',
+                options: {
+                  babelrc: false,
+                  presets: [
+                    BABEL_PRESET_PATH,
+                  ],
+                },
+              },
+            ],
           },
+          ...this.env.postLoaders.map(loader => ({
+            enforce: 'post',
+            ...loader
+          })),
         ],
-        postLoaders: this.env.postLoaders || [],
         noParse: this.env.noParse,
       },
 
       resolve: {
-        extensions: ['.js', '.json', ''],
-        postfixes: [''],
-        modulesDirectories: ['webpackShims', 'node_modules'],
-        fallback: [fromRoot('webpackShims'), fromRoot('node_modules')],
-        loaderPostfixes: ['-loader', ''],
-        root: fromRoot('.'),
+        extensions: ['.js', '.json'],
+        mainFields: ['browser', 'main'],
+        modules: [
+          'webpackShims',
+          'node_modules',
+          fromRoot('webpackShims'),
+          fromRoot('node_modules'),
+        ],
         alias: this.env.aliases,
         unsafeCache: this.unsafeCache,
       },
 
       resolveLoader: {
-        alias: transform(pkg.dependencies, function (aliases, version, name) {
-          if (name.endsWith('-loader')) {
-            aliases[name] = require.resolve(name);
+        alias: Object.keys(pkg.dependencies || {}).reduce((acc, key) => {
```

Comment:
I removed `resolveLoader` entirely and the build succeeded. Not sure if this block is needed anymore? If it's still needed, maybe add a comment that clarifies its purpose?

Reply:
You're right, I couldn't find the docs for this at first but webpack 2 changed the way loaders are resolved in the module rules: https://webpack.js.org/configuration/module/#useentry

In webpack 1 the loaders defined here were resolved relative to the file they were going to load, which meant that plugins in other projects could accidentally overwrite the loaders Kibana was trying to use, which is why the aliases were used to enforce proper resolution.

In webpack 2 loaders are now resolved relative to the webpackConfig.context, which is set to the root of the Kibana repo.